File: src/main/java/net/heroslender/DataBase/Storage.java
package net.heroslender.DataBase;

import net.heroslender.data.Loja;

/**
 * Created by Heroslender.
 */
public interface Storage {
    void load();

    void unload();

    /**
     * Procurar a loja de um player pelo nome dele
     *
     * @param dono Nome do player
     * @return Loja do player
     */
    Loja get(String dono);

    void save(Loja loja);

    void saveVoto(Loja loja, String votador);

    void delete(Loja loja);
}

File: src/main/kotlin/com/heroslender/viptools/storage/Storage.kt
package com.heroslender.viptools.storage

import com.heroslender.viptools.loja.Loja
import org.bukkit.Location

interface Storage {

    /**
     * Procurar a loja de um player pelo nome dele
     *
     * @param dono Nome do player
     * @return Loja do player
     */
    operator fun get(dono: String): Loja?

    operator fun get(signLocation: Location): Loja?

    fun insert(loja: Loja)

    fun delete(loja: Loja)

    fun updateLocation(loja: Loja)

    fun saveVoto(loja: Loja, votador: String)

    fun deleteVoto(loja: Loja, votador: String)

    fun saveSign(loja: Loja, location: Location)

    fun deleteSign(loja: Loja, location: Location)

    fun onDisable()
}